lib/web_scraper.dart

```diff
@@ -1,167 +1,220 @@
+          if (match != null) {
+            // list for all the occurence of the variable name
+            var temp = result[variableName];
+            if (result[variableName] == null) {
+              temp = [];
+            }
+            temp.add(script.text.substring(match.start, match.end));
+            result[variableName] = temp;
+          }
+        });
+      }
+    }
+
+    // Returning final result i.e. Map of variable names with the list of their occurences.
+    return result;
+  }
+
+  /// Returns webpage's html in string format.
+  String getPageContent() => _response != null
+      ? _response.body.toString()
+      : throw WebScraperException(
+          'ERROR: Webpage need to be loaded first, try calling loadWebPage');
+
+  /// Returns List of elements titles found at specified address.
+  /// Example address: "div.item > a.title" where item and title are class names of div and a tag respectively.
+  /// For ease of access, when using Chrome inspection tool, right click the item you want to copy, then click "Inspect" and at the console, right click the highlighted item, right click and then click "Copy > Copy selector" and provide as String address parameter to this method.
+  List<String> getElementTitle(String address) {
+    if (_response == null) {
+      throw WebScraperException(
+          'getElement cannot be called before loadWebPage');
+    }
+    // Using html parser and query selector to get a list of particular element.
+    var elements = document.querySelectorAll(address);
+    // ignore: omit_local_variable_types
+    List<String> elementData = [];
+
+    for (var element in elements) {
+      // Checks if the element's text is null before adding it to the list.
+      if (element.text != null) {
+        elementData.add(element.text);
+      }
+    }
+    return elementData;
+  }
+
+  /// Returns List of elements' attributes found at specified address respecting the provided attribute requirement.
+  /// Example address: "div.item > a.title" where item and title are class names of div and a tag respectively.
+  /// For ease of access, when using Chrome inspection tool, right click the item you want to copy, then click "Inspect" and at the console, right click the highlighted item, right click and then click "Copy > Copy selector" and provide as String parameter to this method.
+  /// Attributes are the bits of information between the HTML tags.
+  /// Per example in <div class="strong and bold" style="width: 100%;" title="Fierce!">
+  /// The element would be "div.strong.and.bold" and the possible attributes to fetch would be EIHER "style" OR "title" returning with EITHER of the values "width: 100%;" OR "Fierce!" respectively.
+  /// To retrieve multiple attributes at once from a single element, please use getElement() instead.
+  List getElementAtribute(String address, String attrib) {
+    // Attribs are the list of attributes required to extract from the html tag(s) ex. ['href', 'title'].

```

Comment:
That's a good explanation already. I would suggest you fix the following things:

- A typo at line 159 it should be `For`.
- In the return type you are specifying List but actually, it should be the exact datatype like in line 136.

Reply:
Fixed both of those. I was under the impression you used dynamic before and since i don't know all the ins and outs of HTML, assumed there could be some cases where ints or doubles would be retrieved, so I left that way. Changed now though!